hack/generator/pkg/codegen/code_generator.go

```diff
@@ -70,13 +71,14 @@ func (generator *CodeGenerator) Generate(ctx context.Context) error {
 	defs := make(astmodel.Types)
 	for i, stage := range generator.pipeline {
 		klog.V(0).Infof("Pipeline stage %d/%d: %s", i+1, len(generator.pipeline), stage.description)
-		updatedDefs, err := stage.Action(ctx, defs)
+		// Defensive copy so that we can compare types in vs out
+		defsOut, err := stage.Action(ctx, defs.Copy())
```

Comment:
Yeah, it's a defensive copy because some of our stages modify the incoming set of values. Rather than allowing that to become a bug, I figured it was better to ensure it was never a problem.

Reply:
Yeah, agree - resolving this.